Explorer/Assets/DCL/Backpack/BackpackGridController.cs

```diff
@@ -4,43 +4,61 @@
 using DCL.AssetsProvision;
 using DCL.AvatarRendering.Wearables.Components;
 using DCL.AvatarRendering.Wearables.Components.Intentions;
+using DCL.AvatarRendering.Wearables.Helpers;
 using DCL.Backpack.BackpackBus;
+using DCL.Backpack.Breadcrumb;
+using DCL.UI;
 using DCL.Web3.Identities;
 using ECS.Prioritization.Components;
 using ECS.StreamableLoading.Common;
+using System;
 using System.Collections.Generic;
 using System.Threading;
-using UnityEngine;
 using UnityEngine.Pool;
 using Utility;
-using ParamPromise = ECS.StreamableLoading.Common.AssetPromise<DCL.AvatarRendering.Wearables.Components.IWearable[], DCL.AvatarRendering.Wearables.Components.Intentions.GetWearableByParamIntention>;
+using Object = UnityEngine.Object;
+using ParamPromise = ECS.StreamableLoading.Common.AssetPromise<DCL.AvatarRendering.Wearables.Helpers.WearablesResponse, DCL.AvatarRendering.Wearables.Components.Intentions.GetWearableByParamIntention>;
 
 namespace DCL.Backpack
 {
     public class BackpackGridController
     {
-        private const string PAGE_NUMBER = "pageNumber";
+        private const string PAGE_NUMBER = "pageNum";
         private const string PAGE_SIZE = "pageSize";
+        private const string CATEGORY = "category";
+        private const string ORDER_BY = "orderBy";
+        private const string COLLECTION_TYPE = "collectionType";
+        private const string ORDER_DIRECTION = "direction";
+        private const string SEARCH = "name";
+        private const string ASCENDING = "ASC";
+        private const string DESCENDING = "DESC";
+        private const string ON_CHAIN_COLLECTION_TYPE = "on-chain";
 
         private const int CURRENT_PAGE_SIZE = 16;
         private static readonly string CURRENT_PAGE_SIZE_STR = CURRENT_PAGE_SIZE.ToString();
 
         private readonly BackpackGridView view;
         private readonly BackpackCommandBus commandBus;
-        private readonly BackpackEventBus eventBus;
         private readonly IWeb3IdentityCache web3IdentityCache;
         private readonly NftTypeIconSO rarityBackgrounds;
         private readonly NFTColorsSO rarityColors;
         private readonly NftTypeIconSO categoryIcons;
         private readonly IBackpackEquipStatusController backpackEquipStatusController;
 
+        private readonly PageSelectorController pageSelectorController;
+        private readonly Dictionary<string, BackpackItemView> usedPoolItems;
```

Comment:
There is something fishy about this dict: you use it both for `URN` keys and `int` converted to `string`. Consider revising

Reply:
There is a complex logic to handle the state of the grid when loading, UI requirements require a full grid of elements loading even if the results are less than the full grid, the int to string is in order to ensure the correct ordering and usage of the elements in the grid once loaded